w3lib/url.py

```diff
@@ -200,6 +200,18 @@ def url_query_cleaner(url, parameterlist=(), sep='&', kvsep='=', remove=False, u
     return url
 
 
+def _add_or_replace_parameters(url, params):
+    parsed = urlsplit(url)
+    args = parse_qsl(parsed.query, keep_blank_values=True)
+
+    new_args = OrderedDict(args)
+    new_args.update(params)
+    new_args = [(k, v) for k, v in new_args.items()]
```

Comment:
Can ``new_args.items()`` be passed to ``urlencode`` directly?

Reply:
I went through the `urlencode` code and it looks like it accepts a dict argument directly. I've removed the loop(dict to tuple).

cc: @lopuhin

Reply:
@nasirhjafri oh nice, thanks! I didn't realise that ``urlencode`` so picky about the first argument, thanks for resolving this nicely.